config/webpack/entries/main.js

```diff
@@ -34,7 +34,7 @@ function importComponents(r) {
   r.keys().forEach(path => {
     const Component = r(path).default;
     const name = extractComponentNameFromPath(path);
-    if (name !== Component.name && process.env.NODE_ENV !== 'production') {
+    if (![name, '_class'].includes(Component.name) && process.env.NODE_ENV !== 'production') {
```

Comment:
I guess the `_class` as a component name comes from the HOC, which is nameless, does it?

FYI there seems to be [a convention with example](https://reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging) for setting a `displayName` of a HOC. However it does not fit the convention for `Component.name` here.

I've also found an old hack which we used to rename `withTranslation(s?)` HOC from `react-i18next` to fit exactly this check. It's not used anymore, because we replaced the HOC with `useTranslation(?s)` hook from the same library. The hack goes as follows:

```jsx
/*


...............A definition of Avatar component here...........................


 */

const AvatarHOC = withNamespaces('user')(Avatar);

AvatarHOC.propTypes = {
  user: PropTypes.object.isRequired,
  size: PropTypes.number,
  source: PropTypes.string,
  link: PropTypes.bool
};

Avatar.propTypes = {
  ...AvatarHOC.propTypes,
  t: PropTypes.func.isRequired
};
// |||||||||||||||||| this is what i mean
// VVVVVVVVVVVVVVVVVV
Object.defineProperty(AvatarHOC, 'name', { value: 'Avatar' });

export default AvatarHOC;
```
Maybe it will be useful for you, perhaps to include in the HOC; so the `Component.name` would be more meaningful.
Anyways this will eventually not matter, as we remove the angularjs completely. And we may just want to stick with [the convention linked above](https://reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging).
I also see this is still a draft PR, so this is just FYI. :slightly_smiling_face:

Reply:
>FYI there seems to be a convention with example for setting a displayName of a HOC. However it does not fit our convention for Component.name here.

Yep, on todo, I think we'll just need first [`babel-plugin-proposal-class-properties`](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties).